src/video_core/renderer_opengl/renderer_opengl.cpp

```diff
@@ -150,6 +151,14 @@ void RendererOpenGL::SwapBuffers() {
     {
         auto aggregator = Common::Profiling::GetTimingResultsAggregator();
         aggregator->AddFrame(profiler.GetPreviousFrameResults());
+        if (VideoCore::g_toggle_framelimit_enabled) {
+            Common::Profiling::Duration frame_time =
+                aggregator->GetAggregatedResults().frame_time.avg;
+            using LongMs = std::chrono::duration<long, std::chrono::microseconds::period>;
+            std::chrono::microseconds average_frame_time =
+                std::chrono::microseconds(std::chrono::duration_cast<LongMs>(frame_time).count());
+            FrameLimiter(average_frame_time, std::chrono::microseconds(1000000) / 60);
+        }
```

Comment:
That doesn't mathematically doesn't make any sense. anything better than microseconds should still give roughly the same result. If milliseconds actually work better, the sleep duration is too long (which is likely, given how most sleep implementations have resolution of about 1ms [meaning you either sleep 16 or 17 ms; but not 16.666ms]).

This is something that can not be fixed by any float math by the way. And the current interface is a lot cleaner (and accurate) in my opinion.